### DISCUSSION TITLE
[BUG] Teleport V10.0.2 handshake error

### DISCUSSION THREAD

**Discussion by rogerioo** [gravitational/teleport#14885] [General]
I've spent two whole days to figure it out why all the suddenly I could not use anymore `ssh user@[node name].[cluster name]` as described [here](https://goteleport.com/docs/server-access/guides/vscode/#step-13-first-time-setup). Every time I try to do, I received the following error:

```bash
ERROR: ssh: handshake failed: ssh: principal "192.168.200.47" not in the set of valid principals for given certificate:
["55f9d86f-8220-461e-afb8-599954fd33a5.teleport.enterprise.com" "55f9d86f-8220-461e-afb8-599954fd33a5"
 "teleport.enterprise.com.teleport.enterprise.com" "teleport.enterprise.com" "localhost" "127.0.0.1" "::1"
 "remote.kube.proxy.teleport.cluster.local"]

kex_exchange_identification: Connection closed by remote host
```

After a lot o debugging, I've noticed that my client was dropping the key exchange and I don't actually know why. My proxy address is `teleport.enterprise.com` and is hosted in a closed server on a VPN. I think that when the teleport server returned to my client the certs, It came with the address that I've mentioned before. However, when I tried to do the `SSH` the server returned It's actual address `192.168.200.47` and I've got this error.

The only solution I've got was to downgrade the package of my client from `v10.0.2` to `v10.0.1` and It works perfectly fine. My teleport server is on `v9.2.3`.

**Comment by webvictim** [comment]
This appears to be the same bug as https://github.com/gravitational/teleport/issues/14743

**Reply by rogerioo** [reply]
Yeah, It is!
